modules/db/src/main/scala/gem/dao/composite/TaggedCoproduct.scala

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.dao.composite
+
+import cats._
+import cats.implicits._
+import doobie._
+import shapeless._
+import shapeless.ops.coproduct.Inject
+
+object TaggedCoproduct {
```

Comment:
This is what took so long. It's simple in the end but took me a while to figure out how to do it.